Contents/Code/__init__.py

```diff
@@ -603,25 +686,136 @@ def update(self, metadata, media, lang):
+
+                                            if rating_info['post_process'] == "round_1":
+                                                add_rating_value = round(add_rating_value, 1) # display score in plot as max=10.0
+                                                rating_value = str(add_rating_value)
+                                            elif rating_info['post_process'] == "int_times_10": # display score in plot as max=100
+                                                add_rating_value = round(add_rating_value, 1)
+                                                rating_value = str(int(round(float(add_rating_value * 10), 0)))
+                                            else:
+                                                rating_value = str(add_rating_value)
+                                            log.debug("Rating value: " + rating_value)
+                                            
+                                            if rating_info['type'] == 'critic' and critic_default_found == False:
+                                                critic_ratings_found += 1
+                                                critic_score_total += add_rating_value
+                                                
+                                                if rating_default == True: # use default provider for rating
+                                                    critic_default_found = True
+                                                    log.debug("Critic Default rating set, will not average scores")
+                                                else: # use average score for rating
+                                                    add_rating_value = round(float(critic_score_total / critic_ratings_found), 1)
+                                                    log.debug("Average Critic Score: " + str(add_rating_value))
+                                                
+                                                # use image from default or rating with most votes
+                                                if (add_votes > critic_votes or rating_default == True) and rating_info['image_good'] and rating_info['image_bad']:
+                                                    if add_rating_value >= rating_info['score_good']:
+                                                        metadata.rating_image = rating_info['image_good']
+                                                    else:
+                                                        metadata.rating_image = rating_info['image_bad']
+                                                
+                                                metadata.rating = add_rating_value
+                                                metadata.rating_count = add_votes
+
+                                                if audience_ratings_found == 0:
```

Comment:
Unfortunately the latest versions of Plex made this custom importer useless due to Plex server constantly crashing during import. I've since switched to jellyfin. I don't have an easy way to spin this back up and test against my library, apologies.

Reply:
That's ok :)

Reply:
I'll see if I can make it work myself on my own fork :)

Thanks for the idea and groundwork :D

Reply:
But, if you have the time.
Could you entertain me what the expected behavior of this lump of code was? 
Can you remember what your thought-process was? :)

Reply:
I believe it was to be sure to always have both critic and audience ratings values set to make sorting work better in the app. If it was 0.0 value it made filtering and sorting hard.